rules-reviewed/openshift/mail.windup.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0"?>
+<ruleset id="mail"
+    xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This is a ruleset for detecting Mail API usage when migrating an application to a cloud environment.
+        </description>
+        <dependencies>
+          <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final"/>
+          <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final"/>
+        </dependencies>
+        <sourceTechnology id="java"/>
+        <sourceTechnology id="java-ee"/>
+        <targetTechnology id="cloud-readiness"/>
+        <tag>mail</tag>
+    </metadata>
+    <rules>
+        <rule id="mail-00000">
+            <when>
+                    <javaclass references="javax.mail.{*}">
+                        <location>CONSTRUCTOR_CALL</location>
+                        <location>IMPORT</location>
+                        <location>INHERITANCE</location>
+                        <location>VARIABLE_DECLARATION</location>
+                    </javaclass>
+            </when>
+            <perform>
+              <hint title="JavaMail: direct access of an SMTP server" effort="3" category-id="cloud-optional">
```

Comment:
@Maarc - AFAIK, JavaMail only supports SMTP for outbound mail communication. It feels weird to me to say that it is about outbound mail, but not SMTP. :) But maybe you really meant that it is about mail server communication in general instead of just about outbound mail.

If that is the case, then perhaps the title can be generic "JavaMail: Direct Access of a mail server dependency". Then the guidance can be something like:

"Verify that the mail connection settings are configured to point to a properly configured backing service"
 - [The Twelve Factor App - Backing Services](https://12factor.net/backing-services)